core/src/worker/mod.rs

```diff
@@ -295,15 +302,18 @@ impl Worker {
         // Unregister worker from current client, register in new client at the end
         let mut worker_key = self.worker_key.lock();
         let slot_provider = (*worker_key).and_then(|k| self.client.workers().unregister(k));
-        self.client
-            .replace_client(super::init_worker_client(&self.config, new_client));
+        self.client.replace_client(super::init_worker_client(
```

Comment:
Can you confirm if I replace a worker's client, it is this new client that will be used for its successive heartbeats? For most use cases of this call, the older client's auth/mTLS is invalid, and if you keep using it, you'll start hitting permission denials and failures. Also, you should not assume if someone replaces a single worker's client, that it can be the replacement for all worker's clients that happen to share the same endpoint and namespace.

This is another reason that workers should use their clients to make heartbeat calls.